src/safeposix/syscalls/net_calls.rs

```diff
@@ -2911,34 +2972,52 @@ impl Cage {
             None => interface::RustDuration::MAX,
         };
 
+        // according to standard, we should clear all revents
+        for structpoll in &mut *fds {
+            structpoll.revents = 0;
+        }
+
+        // we loop until either timeout
+        // or any of the file descriptor is ready
         loop {
+            // iterate through each file descriptor
             for structpoll in &mut *fds {
+                // get the file descriptor
                 let fd = structpoll.fd;
+
+                // according to standard, we should ignore all file descriptor
```

Comment:
How can file descriptors be less than 0?

Reply:
a normal file descriptor could not be less than 0 of course. But the fd field is an user input, so it could be filled with anything that user may pass. And actually passing negative fd here in poll_syscall has a very common use, that it allows user to temporary disable monitoring this fd by just setting fd to -1.